### PR TITLE
Implement Aodh salt formula

### PR DESCRIPTION
**Pull request by ityaptin**
This CR adds following things:
- Configs and sls scripts for Aodh packages installation
- aodh.conf template and parameters
- Heka config for aodh logs parsing

### CODE REVIEW

**Review by cznewt — CHANGES_REQUESTED** [review]
Make config files per openstack version as other openstack services have

**Comment by ityaptin** [comment]
@cznewt Hello! Aodh has been moved to separate project from Ceilometer and alarming code has been removed from Ceilometer only during Mitaka. So, from my PoV Aodh shouldn't be deployed for the earliest releases and shouldn't have a configs for it too.

File: aodh/map.jinja [hunk 1]
```diff
@@ -1,29 +1,11 @@
...
+{% set server = salt['grains.filter_by']({
+    'Debian': {
+        pkgs: ['aodh-api', 'aodh-evaluator', 'aodh-notifier', 'aodh-listener', 'python-aodhclient'],
+        services: ['aodh-api', 'aodh-evaluator', 'aodh-notifier', 'aodh-listener']
+    },
+    'RedHat': {
+        pkgs: ['openstack-aodh-api', 'openstack-aodh-evaluator', 'openstack-aodh-notifier', 'openstack-aodh-listener', 'python-aodhclient'],
+        services: ['openstack-aodh-api', 'openstack-aodh-evaluator', 'openstack-aodh-notifier', 'openstack-aodh-listener']
+    },
+}, merge=salt['pillar.get']('ceilometer:server')) %}
```
Comment (simonpasquier) [inline aodh/map.jinja]: s/ceilometer:server/aodh:server/

**Review by simonpasquier — APPROVED** [review]

**Review by simonpasquier — CHANGES_REQUESTED** [review]
one minor comment from me.

File: aodh/map.jinja (hunk 1 above)
Comment (ityaptin) [reply aodh/map.jinja]: I'll fix it asap

File: aodh/server.sls [hunk 2]
```diff
@@ -1,4 +1,22 @@
 {%- from "aodh/map.jinja" import server with context %}
-{%- if server.enabled %}
+{%- if server.enabled and server.version == "mitaka" %}
```
Comment (simonpasquier) [inline aodh/server.sls]: is it a common pattern for other OpenStack formulas to check the version?

Comment (ityaptin) [reply aodh/server.sls]: No, {%- if server.enabled %} is common pattern, but Aodh is supported only still Mitaka and there is not Aodh on earlier openstack versions.

Comment (simonpasquier) [reply aodh/server.sls]: Indeed. I was wondering about 'server.version == "mitaka"'. Adding this test will force you to update it for every new OpenStack release.

File: aodh/server.sls [hunk 3]
```diff
@@ -1,4 +1,25 @@
 {%- from "aodh/map.jinja" import server with context %}
 {%- if server.enabled %}
+# Exclude unsupported openstack versions
+{%- if server.version >= "mitaka"  %}
```
Comment (simonpasquier) [inline aodh/server.sls:4]: I wouldn't even mind about testing this. If you try to apply the aodh roles to a liberty (or earlier) environment, it should crash. But maybe @cznewt has other views...

**Review by simonpasquier — APPROVED** [review]

File: aodh/files/heka.toml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+file_match = '(?P<Service>.+)\.log\.?(?P<Index>\d*)?(.gz)?'
+differentiator = ["'aodh','_','Service'"]
+priority = ["^Index"]
+decoder = "openstack"
+oldest_duration = "168h"
+
+[openstack]
+type = "SandboxDecoder"
+filename = "lua_modules/decoders/openstack.lua"
+module_directory = "/usr/share/heka/lua_modules;/usr/share/heka/lua_modules/common"
```
Comment (elemoine) [inline aodh/files/heka.toml]: You should not have this file. All the Heka configuration related to Aodh should be in `aodh/meta/heka.yml`.

File: aodh/meta/salt.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+orchestrate:
+  server:
+    priority: 690
+    batch: 1
+    require:
+    - salt: keystone.server
```
Comment (elemoine) [inline aodh/meta/salt.yml]: Do we want that? I am not sure the `orchestrate` meta is still a thing… @cznewt?

File: aodh/meta/sphinx.yml [hunk 6]
```diff
@@ -1,12 +1,39 @@
-{%- from "aodh/map.jinja" import server with context %}
-# Fill in documentation details
 doc:
-  name: aodh
-  description: Some service info
+  name: Aogh
```
Comment (elemoine) [inline aodh/meta/sphinx.yml]: s/Aogh/Aodh?

File: metadata/service/support.yml [hunk 7]
```diff
@@ -4,7 +4,7 @@ parameters:
       collectd:
         enabled: false
       heka:
-        enabled: false
+        enabled: true
```
Comment (elemoine) [inline metadata/service/support.yml]: You have to have a `meta/heka.yml` if you enable Heka support. For now I think I would focus on Aodh itself, and defer the support of Heka.

**Review by elemoine — CHANGES_REQUESTED** [review]

File: metadata/service/support.yml (hunk 7 above)
Comment (ityaptin) [reply metadata/service/support.yml]: Ok, I agree!

File: aodh/meta/sphinx.yml (hunk 6 above)
Comment (ityaptin) [reply aodh/meta/sphinx.yml]: done

File: aodh/files/heka.toml (hunk 4 above)
Comment (ityaptin) [reply aodh/files/heka.toml]: Deleted

File: aodh/meta/salt.yml (hunk 5 above)
Comment (ityaptin) [reply aodh/meta/salt.yml]: You are right. Deleted.

**Pull request merged**